swagger/ledger.yaml

```diff
@@ -65,6 +130,254 @@ definitions:
+        description: json formatted string holding dynamic data
+        example: {note: lorem ipsum}
+
+  sub_category:
+    type: object
+    properties:
+      balance:
+        type: integer
+        description: Amount in cents.
+        example: -66534
+      count:
+        type: integer
+        description: Number of transactions in this category.
+        example:
+          min:
+            value: 0
+      credit:
+        type: integer
+        description: Total credit amount.
+        example: 7
+      credit_count:
+        type: integer
+        description: Number of credit transactions
+        example:
+          min:
+            value: 0
+      debit:
+        type: integer
+        description: Total debit amount.
+        example: -454921
+      debit_count:
+        type: integer
+        description: Number of debit transactions
+        example:
+          min:
+            value: 0
+            
+  balance:
+    type: object
+    properties:
+      project_id:
+        type: string
+        description: UUID of the project.
+        example: c9e0e14a-17d2-4250-96d0-e95df41cc0c3
+      total_balance:
+        type: integer
+        description: Amount in cents.
+        example: 1503428
+      available_balance:
+        type: integer
+        description: Available balance for the project. totalBalance - feeBalance.
+        example: 1488394
+      fee_balance:
+        type: integer
+        description: Total unpaid fees.
+        example: 15034
+      total_credit:
+        type: integer
+        description: Total credit to project.
+        example: 2503428
+      total_debit:
+        type: integer
+        description: Total debit from project.
+        example: -1000000
+      total_count:
+        type: integer
+        description: Credit transaction count.
+        example:
+          min:
+            value: 0
+      credit_count:
+        type: integer
+        description: Credit transaction count.
+        example:
+          min:
+            value: 0
+      debit_count:
+        type: integer
+        description: Debit transaction count.
+        example:
+          min:
+            value: 0
+      backers:
+        type: integer
+        description: Total backers for the project.
+        example:
+          min:
+            value: 0
+      subTotals:
+        properties:
+          BugBounty:
+            $ref: '#/definitions/sub_category'
+          Development:
+            $ref: '#/definitions/sub_category'
+          Documentation:
+            $ref: '#/definitions/sub_category'
+          Marketing:
+            $ref: '#/definitions/sub_category'
+          Meetups:
+            $ref: '#/definitions/sub_category'
+          Mentee:
+            $ref: '#/definitions/sub_category'
+          Travel:
+            $ref: '#/definitions/sub_category'
+          Other:
+            $ref: '#/definitions/sub_category'
+          Uncategorised:
+            $ref: '#/definitions/sub_category'
```

Comment:
Instead of listing out all the "categories", make this a dynamic list (array of categories).

Reply:
Changed to array of items.